AppCenterIdentity/AppCenterIdentity/Internals/Models/MSIdentityConfig.m

```diff
@@ -0,0 +1,57 @@
+#import "MSIdentityConfig.h"
+
+@implementation MSIdentityConfig
+
+static NSString *const kMSIdentityScope = @"identity_scope";
+
+static NSString *const kMSClientId = @"client_id";
+
+static NSString *const kMSRedirectUri = @"redirect_uri";
+
+static NSString *const kMSAuthorities = @"authorities";
+
+static NSString *const kMSAuthorityTypeB2C = @"B2C";
+
+- (instancetype)initWithDictionary:(NSDictionary *)dictionary {
+  if (!dictionary) {
+    return nil;
+  }
+  if ((self = [super init])) {
+    if (dictionary[kMSIdentityScope]) {
+      self.identityScope = (NSString * _Nonnull) dictionary[kMSIdentityScope];
+    }
+    if (dictionary[kMSClientId]) {
+      self.clientId = (NSString * _Nonnull) dictionary[kMSClientId];
+    }
+    if (dictionary[kMSRedirectUri]) {
+      self.redirectUri = (NSString * _Nonnull) dictionary[kMSRedirectUri];
+    }
+    if (dictionary[kMSAuthorities]) {
+      NSMutableArray *array = [NSMutableArray new];
+      for (NSDictionary *authorityDic in dictionary[kMSAuthorities]) {
+        [array addObject:[[MSIdentityAuthority alloc] initWithDictionary:authorityDic]];
+      }
+      self.authorities = array;
+    }
+  }
+  return self;
+}
+
+- (BOOL)isValid {
+  return self.identityScope && self.clientId && self.redirectUri && [self areAuthoritiesValid];
+}
+
+- (BOOL)areAuthoritiesValid {
+  BOOL seenDefault = NO;
```

Comment:
Nit: can we call this local variable `foundDefault` or something? I was confused for a minute reading this (I think using the word "seen" is a bit too much personification for my taste)